### ISSUE TITLE
Customer wants removed Clone() re-added to C# dots objects

### ISSUE THREAD

**Issue by DonOregano** [SafirSDK/safir-sdk-core#232]
In 6.0 we removed the Clone() function from dots objects in C# and Java, since it was no longer needed internally. Unfortunately our users have been using the Clone function quite extensively and would like it re-added.

**Comment by DonOregano** [comment]
A solution was made using the DeepClone workaround code that was included in 6.0, but instead of just providing the DeepClone.cs file and have the user add it to their projects we've used it to provide the removed Clone() interface.

DeepClone has been added to dots_dotnet.ss, and a Clone method re-added to Safir.Dob.Typesystem.Object.

The difference between the original (before 6.0) Clone and this Clone is that the previous version consisted of thousands of lines of code, whereas this solution uses C# reflection and is a couple of hundred lines.

Tests for the new Clone have been added to the dotnet dots tests.
